backend/remote-state/gcs/client.go

```diff
@@ -90,19 +148,21 @@ func (c *remoteClient) Lock(info *state.LockInfo) (string, error) {
 	}
 
 	lockFile := c.lockFile()
-	w := lockFile.If(storage.Conditions{DoesNotExist: true}).NewWriter(c.storageContext)
-	err = func() error {
-		if _, err := w.Write(infoJson); err != nil {
-			return err
-		}
-		return w.Close()
-	}()
 
+	generation, err := c.createLockFile(lockFile, infoJson)
 	if err != nil {
-		return "", c.lockError(fmt.Errorf("writing %q failed: %v", c.lockFileURL(), err))
+		if c.unlockIfStale(lockFile) {
```

Comment:
I'd like to see a log warning line about unlocking an unexpected lock. (Hopefully we'll get better log handling soon, and these will be displayed to the user)

Reply:
I've added some log output for this case to unlockIfStale().